apollo-integration/src/test/java/com/apollographql/apollo/ResponseWriteTestCase.java

```diff
@@ -356,4 +357,79 @@ public void operation_with_inline_fragments() throws Exception {
     assertThat(hero.friends().get(1).asDroid().name()).isEqualTo("RD");
     assertThat(hero.friends().get(1).asDroid().primaryFunction()).isEqualTo("Entertainment");
   }
+
+  @Test
+  public void fragments() throws Exception {
+    server.enqueue(mockResponse("HeroAndFriendsWithFragmentResponse.json"));
+
+    HeroAndFriendsWithFragments query = new HeroAndFriendsWithFragments(Episode.NEWHOPE);
+    HeroAndFriendsWithFragments.Hero hero = apolloClient.query(query).execute().data().hero();
+
+    assertThat(hero.__typename()).isEqualTo("Droid");
+    assertThat(hero.fragments().heroWithFriendsFragment().__typename()).isEqualTo("Droid");
+    assertThat(hero.fragments().heroWithFriendsFragment().id()).isEqualTo("2001");
+    assertThat(hero.fragments().heroWithFriendsFragment().name()).isEqualTo("R2-D2");
+    assertThat(hero.fragments().heroWithFriendsFragment().friends()).hasSize(3);
+    assertThat(hero.fragments().heroWithFriendsFragment().friends().get(0).fragments().humanWithIdFragment().__typename()).isEqualTo("Human");
+    assertThat(hero.fragments().heroWithFriendsFragment().friends().get(0).fragments().humanWithIdFragment().id()).isEqualTo("1000");
+    assertThat(hero.fragments().heroWithFriendsFragment().friends().get(0).fragments().humanWithIdFragment().name()).isEqualTo("Luke Skywalker");
+    assertThat(hero.fragments().heroWithFriendsFragment().friends().get(1).fragments().humanWithIdFragment().__typename()).isEqualTo("Human");
+    assertThat(hero.fragments().heroWithFriendsFragment().friends().get(1).fragments().humanWithIdFragment().id()).isEqualTo("1002");
+    assertThat(hero.fragments().heroWithFriendsFragment().friends().get(1).fragments().humanWithIdFragment().name()).isEqualTo("Han Solo");
+    assertThat(hero.fragments().heroWithFriendsFragment().friends().get(2).fragments().humanWithIdFragment().__typename()).isEqualTo("Human");
+    assertThat(hero.fragments().heroWithFriendsFragment().friends().get(2).fragments().humanWithIdFragment().id()).isEqualTo("1003");
+    assertThat(hero.fragments().heroWithFriendsFragment().friends().get(2).fragments().humanWithIdFragment().name()).isEqualTo("Leia Organa");
+
+    apolloClient.apolloStore().write(
+        new HeroWithFriendsFragment(
```

Comment:
thoughts on a builder? With more than 3 params it would be tough to see what corresponds to what.